### ISSUE TITLE
Multi-GPU size mismatch

### ISSUE THREAD

**Issue by xubenben** [salesforce/awd-lstm-lm#12]
I try to train lm with multi-GPU(In this case, I use 3 GPU) by set "model = torch.nn.DataParallel(model).cuda()", and change "model.init_hidden" to "model.module.init_hidden", then meet error:

<img width="425" alt="2017-11-09 4 42 48" src="https://user-images.githubusercontent.com/3480189/32596088-b74d3fca-c4f7-11e7-9c78-d033bcd926d9.png">

It seems that only one GPU's result has been collected, I can't explain what happened.

**Comment by xubenben** [comment]
Maybe I should use "model = torch.nn.DataParallel(model, dim=1).cuda()"
